client/connect-account-page/index.js

```diff
@@ -96,7 +100,50 @@ const ConnectPageOnboardingDisabled = () => (
 const ConnectPageOnboarding = () => {
 	const [ isSubmitted, setSubmitted ] = useState( false );
 
-	const handleSetup = () => {
+	const handleLocationCheck = ( availableCountries ) => {
+		// Reset the 'Set up' button state if merchant decided to stop
+		const whenDeclined = () => {
+			setSubmitted( false );
+		};
+		// Redirect the merchant if merchant decided to continue
+		const whenConfirmed = () => {
+			window.location = wcpaySettings.connect.url;
+		};
+
+		// Populate translated list of supported countries we want to render in the modal window.
+		const countries = Object.values( availableCountries )
+			.map( ( country ) => {
+				// eslint-disable-next-line @wordpress/i18n-no-variables
+				return __( country, 'woocommerce-payments' );
```

Comment:
I believe the `__()` function only works with string literals.

e.g.

```php
// This is incorrect do not use.
_e( "Your city is $city.", 'my-theme' );
```

>The strings for translation are extracted from the source without executing the PHP associated with it. For example: The variable $city may be Vancouver, so your string will read "Your city is Vancouver" when the template is run but gettext won’t know what is inside the PHP variable in advance.

>As the value of the variable is unknown when your string is translated, it would require the translator to know every case for the variable $country. This is not ideal, and it is best to remove dynamic content and allow translators to focus on static content.

More info here - https://developer.wordpress.org/apis/handbook/internationalization/internationalization-guidelines/#variables

(This reference is for PHP but the same applies to JS)

We can probably wrap the country strings in i18n functions inside `supported_countries()` and they'll be translated by the time they reach JS.

Reply:
Yes, '__' works with string literals. Translations on the backend side is a good option, relocated translation there in 14edd2f807fa702c811a6f6ba5bfcc32cf885398